src/core/features/emulator/services/file-transfer.ts

```diff
@@ -43,40 +40,24 @@ export class FileTransferErrorMock implements FileTransferError {
 /**
  * Emulates the Cordova FileTransfer plugin in desktop apps and in browser.
  */
-@Injectable()
-export class FileTransferMock extends FileTransfer {
-
-    /**
-     * Creates a new FileTransferObjectMock object.
-     *
-     * @returns a new file transfer mock.
-     */
-    create(): FileTransferObjectMock {
-        return new FileTransferObjectMock();
-    }
-
-}
-
-/**
- * Emulates the FileTransferObject class in desktop apps and in browser.
- */
-export class FileTransferObjectMock extends FileTransferObject {
+export class FileTransferMock {
 
     progressListener?: (event: ProgressEvent) => void;
     source?: string;
     target?: string;
     xhr?: XMLHttpRequest;
 
-    protected reject?: (reason?: unknown) => void;
+    onprogress!: (event: ProgressEvent) => void;
```

Comment:
```suggestion
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    onprogress: (event: ProgressEvent) => void = () => {};
```

Instead of using "!" I think it's better to initialize this with a function that does nothing. This way you can always call onprogress in the implementation without having to check if it exists or not. E.g. instead of doing this:

```
if (this.progressListener) {
    this.progressListener(ev);
}
```

Now you will be able to just do:

`this.onprogress(ev);`